### DISCUSSION TITLE
Join data from another table?

### DISCUSSION THREAD

**Discussion by madebydor** [supabase/supabase#3912] [Questions] [answered]
I have the following 3 tables:

| users |         
|----------|    
| id  (uuid)            | 
| team_id (uuid references teams)            | 
|fullName  (text)              |

|teams|     
|----------|
|id  (uuid)             |
|name (text)              |

|teams_users|
|------------------|
|user_id (uuid references users) |
|team_id (uuid references teams) |

How can I display a team and list the names of all its users, of the user who requested the data?

my broken code:

```js
await supabase.from('teams').select(`
  name,
  users(name)
`);
```

That's the result im into:

```js
{
    name: "My beautiful team",
    users:[
        { name: "Jake" },
        { name: "Supabase" }
    ]
}
````

**Comment by fergusmeiklejohn** [comment]
It doesn't seem like `name` is a foreign key referencing anything. Try using the FK that links teams and users.
The Supabase JS library is more difficult to play around with than SQL. To figure out more complex queries, try using a tool like DataGrip (it has a free month trial). You can connect to the database and figure out the SQL queries you want to use.

**Comment by madebydor** [comment, marked as answer]
Here's the solution:
```sql
-- Parameters need to be prefixed because the name clashes with `cm`'s columns
CREATE OR REPLACE FUNCTION is_team_user(_user_id uuid, _company_id uuid) RETURNS bool AS $$
SELECT EXISTS (
  SELECT 1
  FROM company_members cm
  WHERE cm.company_id = _company_id
  AND cm.user_id = _user_id
);
$$ LANGUAGE sql SECURITY DEFINER;
```

and then I create a policy and use the function, like so:
````js
is_team_user(uid(), company_id)
````
